closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1027 @@
+import java.security.Provider.Service;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+import sun.security.util.Debug;
+
+/**
+ * Configures the security providers when in restricted security mode.
+ */
+public final class RestrictedSecurity {
+
+    private static final Debug debug = Debug.getInstance("semerufips");
+
+    // Restricted security mode enable check, only supported on Linux x64.
+    private static final boolean userEnabledFIPS;
+    private static final boolean userEnabledSecurity;
+    private static final boolean isSecuritySupported;
+    private static final boolean shouldEnableSecurity;
+    private static final String userSecuritySetting;
+
+    private static boolean securityEnabled;
+
+    private static int userSecurityNum;
+    private static boolean userSecurityTrace;
+    private static boolean userSecurityAudit;
+    private static boolean userSecurityHelp;
+
+    private static RestrictedSecurityProperties restricts;
+
+    private static final List<String> supportPlatforms = List.of("amd64");
+
+    static {
+        String[] props = AccessController.doPrivileged(
+                new PrivilegedAction<>() {
+                    @Override
+                    public String[] run() {
+                        return new String[] { System.getProperty("semeru.fips"),
+                                System.getProperty("semeru.restrictedsecurity"),
+                                System.getProperty("os.name"),
+                                System.getProperty("os.arch") };
+                    }
+                });
+        userEnabledFIPS = Boolean.parseBoolean(props[0]);
+        // If semeru.fips is true, then ignore semeru.restrictedsecurity, use userSecurityNum 1.
+        userSecuritySetting = userEnabledFIPS ? "1" : props[1];
+        userEnabledSecurity = !isNullOrBlank(userSecuritySetting);
+        isSecuritySupported = "Linux".equalsIgnoreCase(props[2])
+                && supportPlatforms.contains(props[3]);
+        shouldEnableSecurity = (userEnabledFIPS || userEnabledSecurity) && isSecuritySupported;
+    }
+
+    private RestrictedSecurity() {
+        super();
+    }
+
+    /**
+     * Restricted security mode will be enabled when the semeru.restrictedsecurity
+     * system property is set.
+     *
+     * @return true if restricted security is enabled
+     */
+    public static boolean isEnabled() {
+        return securityEnabled;
+    }
+
+    /**
+     * Restricted security mode secure random provider
+     *
+     * @return the secure random provider
+     */
+    public static String getRandomProvider() {
+        return restricts.jdkSecureRandomProvider;
```

Comment:
@keithc-ca 
I added a new method "restrictsCheck()", and in that method, it will check the "restricts". And I also moved the SunsetDate check and SecureRandom check into this "restrictsCheck()" method. To make sure, once the "securityEnabled" is true, then the "restricts" and SecureRandom won't be null.

Reply:
There's also the problem that `configure(Properties)` can be called more than once without error.

Reply:
@keithc-ca 
I added a check in the “configure()” method. If the “securityEnabled” is true when go into this “configure()” method, that means the restricted security mode is already initialized. Then, throw a runtime exception explains the restricted security mode can't be initialized twice.

Reply:
For the checks in "configure()", please see the latest comment.